pac4j-kerberos/src/main/java/org/pac4j/kerberos/credentials/extractor/KerberosExtractor.java

```diff
@@ -0,0 +1,44 @@
+package org.pac4j.kerberos.credentials.extractor;
+
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.extractor.CredentialsExtractor;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.kerberos.credentials.KerberosCredentials;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * To extract Kerberos headers.
+ *
+ * @author Garry Boyce
+ * @since 2.1.0
+ */
+public class KerberosExtractor implements CredentialsExtractor<KerberosCredentials> {
+    private final String clientName;
+
+    public KerberosExtractor(final String clientName) {
+        this.clientName = clientName;
+    }
+
+    @Override
+    public KerberosCredentials extract(WebContext context) throws CredentialsException, HttpAction {
+        final String header = context.getRequestHeader(HttpConstants.AUTHORIZATION_HEADER);
+        if (header == null) {
+            throw HttpAction.unauthorizedNegotiate("Kerberos Header not found", context);
```

Comment:
so, @leleuj , looking at other extractors/clients,  I guess this exception throwing is not the responsibility of `KerberosExtractor` and should be moved to `Kerberos(In)?DirectClient`.

there is one semantic nuance with SPNEGO/Kerberos - the browser (e.g. Firefox) do NOT send the credentials (`Authorization: Negotiate someTicket`) if server did not provide a `WWW-Authenticate: Negotiate` header. 

Thus a trully DirectClient don't quite make sense here. As it is now, the 401 indicates to repeat the request with credentials, but does do a redirect, so it's in between direct and indirect, I guess?

---
in my case, I'd like to combine session/cookie storage with Kerberos, so would not need renegotiate with every request. so in brief what I'd like to have sounds like indirect client.

would this happen automatically if it implemented an indirect client?